docs/troubleshooting_guide.md

```diff
@@ -0,0 +1,358 @@
+```
+
+#### 5. Existing ORACLE_HOME
+
+* Error:
+```bash
+"Launching Oracle Grid Infrastructure Setup Wizard...", "", "[WARNING] [INS-40109] The specified Oracle Base location is not empty on this server.", "   ACTION: Specify an empty location for Oracle Base.", "[WARNING] [INS-32047] The location (/u01/app/oraInventory) specified for the central inventory is not empty.", "   ACTION: It is recommended to provide an empty location for the inventory.", "[FATAL] [INS-13019] Some mandatory prerequisites are not met. These prerequisites cannot be ignored."
+```
+
+* Fix:
+Rerun `cleanup-oracle.sh` and ensure it finished successfully (it can fail at the part where it tries to umount the u01 mountpoint and exits at that point)
+
+#### 6. First time connection errors to the BM from cloud VM
+* Customers will need to generate ssh public and private keys, store the public key under the home directory of the `ansible` OS user in the BM host and the private keys under their OS account in cloud VM.
+* Sample connectivity error may be of the form similar to:
+
+```bash
+TASK [Test connectivity to target instance via ping] ***********************************************************************************************************
+fatal: [racnode1.orcl]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: Warning: Permanently added '172.16.30.1' (ECDSA) to the list of known hosts.\r\nno such identity: /home/janedoe/.ssh/id_rsa: No such file or directory\r\nPermission denied (publickey,gssapi-keyex,gssapi-with-mic,password).", "unreachable": true}
+fatal: [racnode2.orcl]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: Warning: Permanently added '172.16.30.2' (ECDSA) to the list of known hosts.\r\nno such identity: /home/janedoe/.ssh/id_rsa: No such file or directory\r\nPermission denied (publickey,gssapi-keyex,gssapi-with-mic,password).", "unreachable": true}
+```
+
+* Fix:
+Reason for the error is as follows:
+* By, `--instance-ssh-user ansible --instance-ssh-key ~/.ssh/id_rsa` typically provided in the command line to `install-oracle.sh`, we are asking Ansible to connect:
+  * from the current jump host/VM as the user (say) `janedoe`
+  * to the BM host as the OS user called `ansible` 
+  * using the SSH key file in the jump host under the current users home directory's: `~/.ssh/id_rsa`
+
+As noted in [bms documentation](https://github.com/google/bms-toolkit/blob/master/docs/user-guide.md#command-quick-reference-for-rac-deployments
+):
+we want to ensure we can connect to the backend nodes with ssh and sudo to the account we want ansible to escalate to:
+```bash
+ssh ${INSTANCE_SSH_USER:-`whoami`}@${INSTANCE_IP_ADDR_NODE_1} sudo -u root hostname
+ssh ${INSTANCE_SSH_USER:-`whoami`}@${INSTANCE_IP_ADDR_NODE_2} sudo -u root hostname
+```
+
+To enable passwordless ssh connectivity and to set up that connection manually, following actions may be performed:
```

Comment:
Perhaps link to a doc that explains this already?  I don't think there's anything specific to the toolkit or ansible that need to be set up.

Reply:
Done